ui/vic-ui-h5c/vic/src/vic-app/src/app/container-view/container-view.component.ts

```diff
@@ -80,71 +82,82 @@ export class VicContainerViewComponent implements OnInit, OnDestroy {
     public readonly GIGABYTE: number = Math.pow(1024, 3);
+    private storedState: StoredState = {
+        currentFilter: '',
+        currentSort: 'id,asc',
+        currentOffset: 0
+    };
 
     constructor(
         private vmViewService: VicVmViewService,
         private refreshService: RefreshService,
         private gs: GlobalsService,
-        public vicI18n: Vic18nService
-    ) {
+        public vicI18n: Vic18nService,
+        private cdr: ChangeDetectorRef
+    ) { }
+
+    ngOnInit() {
         // subscribes to the global refresh event
         this.refreshSubscription = this.refreshService
             .refreshObservable$.subscribe(() => {
-                this.reloadContainers();
+                this.reloadContainers(this.storedState);
             });
     }
 
-    ngOnInit() {
-        // sets up a listener for updating this.containers
-        this.vmViewService.containers$.subscribe(vms => {
-            this.containers = vms;
-            this.isDgLoading = false;
-        }, err => {
-            this.containers = [];
-            console.error(err);
-        });
-    }
-
     ngOnDestroy() {
-        this.refreshSubscription.unsubscribe();
+        if (this.refreshSubscription) {
+            this.refreshSubscription.unsubscribe();
+        }
     }
 
     /**
      * Queries vic-service with the current Datagrid state
      * @param state current Datagrid state
      */
     refreshGrid(state: State) {
-        this.currentFilter = state.filters ? state.filters
-            .map(item => item['property'] + '=' + item['value'])
-            .join(',') : '';
-
-        if (state.sort) {
-            let sortBy = typeof state.sort.by === 'object' ?
-                state.sort.by.toString() : state.sort.by;
-
-            this.currentSort = `${sortBy},${state.sort.reverse ? 'desc' : 'asc'}`;
+        let newState = processState(state);
+        let oldState = this.storedState;
+        if (oldState) {
+            let sameState =
+                newState['currentFilter'] === oldState['currentFilter'] &&
+                newState['currentSort'] === oldState['currentSort'] &&
+                newState['currentOffset'] === oldState['currentOffset'];
+
+            if (sameState) {
+                // do not fetch data when state has not changed
+                return;
+            }
         }
 
-        this.currentOffset = state.page.from;
-        this.reloadContainers();
+        this.storedState = newState;
+        this.reloadContainers(this.storedState);
     }
 
     /**
      * Calls vm-view service to reload Containers
      */
-    reloadContainers() {
+    reloadContainers(storedState?: StoredState) {
         this.isDgLoading = true;
+
+        let offset = storedState ? storedState.currentOffset : 0;
+        let sort = storedState ? storedState.currentSort : 'id,asc';
+        let filter = storedState ? storedState.currentFilter : '';
+
         this.vmViewService.reloadContainers({
```

Comment:
vmViewService method name needs to reflect that it is only fetching data since it doesn't actually handle refreshing and since it returns an observable then should have $ appended to the name.